src/components/OrgUnitDimension/__tests__/OrgUnitDimension.spec.js

```diff
@@ -1,57 +1,90 @@
-import { shallow } from 'enzyme'
+import { CustomDataProvider } from '@dhis2/app-runtime'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 import React from 'react'
 import OrgUnitDimension from '../OrgUnitDimension.js'
 
-describe('The OrgUnitDimension component', () => {
-    let props
-    let shallowOrgUnitDimension
+jest.mock('@dhis2-ui/organisation-unit-tree', () => {
+    const lib = jest.requireActual('@dhis2-ui/organisation-unit-tree')
 
-    const getWrapper = () => {
-        if (!shallowOrgUnitDimension) {
-            shallowOrgUnitDimension = shallow(<OrgUnitDimension {...props} />)
-        }
-        return shallowOrgUnitDimension
+    return {
+        ...lib,
+        OrganisationUnitTree: () => <div>Org unit tree component mock</div>,
     }
+})
 
-    beforeEach(() => {
-        props = {
-            roots: [],
-            selected: [],
-            onSelect: jest.fn(),
-            hideGroupSelect: false,
-            hideLevelSelect: false,
-            hideUserOrgUnits: false,
-            warning: '',
-        }
-        shallowOrgUnitDimension = undefined
-    })
+describe('OrgUnitDimension', () => {
+    const onSelect = jest.fn()
+
+    const props = {
+        roots: [],
+        selected: [],
+        onSelect: onSelect,
+        hideGroupSelect: false,
+        hideLevelSelect: false,
+        hideUserOrgUnits: false,
+        warning: '',
+    }
+
+    const renderOrgUnitDimension = (props) =>
+        render(
+            <CustomDataProvider
+                data={{
+                    organisationUnitLevels: {
+                        organisationUnitLevels: [],
+                    },
+                    organisationUnitGroups: {
+                        organisationUnitGroups: [],
+                    },
+                }}
+            >
+                <OrgUnitDimension {...props} />
+            </CustomDataProvider>
+        )
+
+    beforeEach(() => onSelect.mockClear())
+
+    test('OrgUnitDimension matches the snapshot', async () => {
+        const { container } = renderOrgUnitDimension(props)
+
+        // avoid the act warning due to the snapshot being taken before async code is run
+        // wait for the component to be loaded, here done by testing that the OrganisationUnitTree component is loaded
+        await screen.findByText('Org unit tree component mock')
 
-    it('matches the snapshot', () => {
-        const wrapper = getWrapper()
-        expect(wrapper).toMatchSnapshot()
+        expect(container).toMatchSnapshot()
```

Comment:
This could potentially cause us headaches with the jsx classes, as mentioned above. We can leave it in for now and if we have to update the snapshots then consider using mocking, or otherwise checking for existing components rather than the snapshot.